dataprofiler/profilers/profile_builder.py

```diff
@@ -2779,6 +2779,34 @@ def _merge_null_replication_metrics(self, other: StructuredProfiler) -> dict:
 
         return merged_properties
 
+    def _auto_multiprocess_toggle(
```

Comment:
Should we do a new function or instead do it in: `suggest_pool_size` which is similar.

Reply:
Also notating this is a staticmethod wrt the class. If we kept the func, maybe should be in profiler_utils

Reply:
An issue I see with doing in `suggest_pool_size` is that it uses a different metric to measure dataset sizes for thresholding (number of rows vs. estimated data size in memory). If we were to do it in there, we would have to change the logic for this (example below). 
```
    if self.options.multiprocess.is_enabled and auto_multiprocess_toggle:
            est_data_size = data[:50000].memory_usage(index=False, deep=True).sum()
            est_data_size = (est_data_size / min(50000, len(data))) * len(data)
            pool, pool_size = profiler_utils.generate_pool(
                max_pool_size=None, data_size=est_data_size, cols=len(data.columns)
            )
```